AndroidIDPTemplate/app/src/com/salesforce/samples/salesforceandroididptemplateapp/MainApplication.kt

```diff
@@ -0,0 +1,73 @@
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * - Neither the name of salesforce.com, inc. nor the names of its contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission of salesforce.com, inc.
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.salesforce.samples.salesforceandroididptemplateapp
+
+import android.app.Application
+
+import com.salesforce.androidsdk.analytics.security.Encryptor
+import com.salesforce.androidsdk.app.SalesforceSDKManager.KeyInterface
+import com.salesforce.androidsdk.smartsync.app.SmartSyncSDKManager
+
+/**
+ * Application class for our application.
+ */
+class MainApplication : Application() {
+
+    override fun onCreate() {
+        super.onCreate()
+        SmartSyncSDKManager.initNative(applicationContext, NativeKeyImpl(), MainActivity::class.java)
+
+        /*
+         * Uncomment the following line to enable IDP login flow. This will allow the user to
+         * either authenticate using the current app or use a designated IDP app for login.
+         * Replace 'idpAppURIScheme' with the URI scheme of the IDP app meant to be used.
+         */
+        // SmartSyncSDKManager.getInstance().idpAppURIScheme = idpAppURIScheme
+
+        /*
+         * Uncomment the following line to enable browser based login. This will use a
+         * Chrome custom tab to login instead of the default WebView. You will also need
+         * to uncomment a few lines of code in SalesforceSDK library project's AndroidManifest.xml.
+         */
+        // SmartSyncSDKManager.getInstance().isBrowserLoginEnabled = true
+
+        /*
+		 * Un-comment the line below to enable push notifications in this app.
+		 * Replace 'pnInterface' with your implementation of 'PushNotificationInterface'.
+		 * Add your Google package ID in 'bootonfig.xml', as the value
+		 * for the key 'androidPushNotificationClientId'.
+		 */
+        // SmartSyncSDKManager.getInstance().pushNotificationReceiver = pnInterface
+    }
+}
+
+internal class NativeKeyImpl : KeyInterface {
```

Comment:
Do we still need this?

Reply:
Not required starting `6.1`, but I left it in intentionally in the templates because we need it for `6.0` for the upgrade steps to get the old key. I left it in here to stay consistent with the templates. I will be marking it deprecated across the board in `6.1` and remove it in `7.0`.